src/main/java/org/opentripplanner/updater/vehicle_parking/VehicleParkingDataSourceFactory.java

```diff
@@ -0,0 +1,23 @@
+package org.opentripplanner.updater.vehicle_parking;
+
+import org.opentripplanner.ext.vehicleparking.kml.KmlBikeParkDataSource;
+import org.opentripplanner.ext.vehicleparking.parkapi.BicycleParkAPIUpdater;
+import org.opentripplanner.ext.vehicleparking.parkapi.CarParkAPIUpdater;
+import org.opentripplanner.routing.vehicle_parking.VehicleParking;
+import org.opentripplanner.updater.DataSource;
+
+public class VehicleParkingDataSourceFactory {
+
+  private VehicleParkingDataSourceFactory() {}
+
+  public static DataSource<VehicleParking> create(VehicleParkingUpdaterParameters source) {
+    switch (source.getSourceType()) {
+      case KML:               return new KmlBikeParkDataSource(source.getUrl(), source.getFeedId(), source.getNamePrefix(), source.isZip());
```

Comment:
Should we create different parameter classes for different updaters that extend the generic parameters similarly as in vehicle rental instead of handpicking parameters from one "VehicleParkingUpdaterParameters" into different data source?

Reply:
We thought it would be better to have different parameter classes for different updaters. We discussed that these parameters shouldn't extend any generic parameter class. However, while doing the changes in https://github.com/opentripplanner/OpenTripPlanner/pull/3796/commits/0901f2f938221f7f647dbd1a2e64557f031df6aa, I ended up implementing it so it extends a generic VehicleRentalUpdaterParameters because all the updaters implement PollingGraphUpdaterParameters and have a data source enum value.